### DISCUSSION TITLE
ECG simulator - PQRST complex

### DISCUSSION THREAD

**Discussion by nachoXXL** [neuropsychology/NeuroKit#556] [Q&A]
Is there a way to determine the PQRST complex from within the ECG simulator, instead of analysing the generated ECG data with ecg_peaks and ecg_delineate? One reason would be to also obtain the onset/offset for Q and S. Another would be that on higher HR's ecg_delineate won't give usable results.
We're looking into doing post-processing of the ECG-signal based on the PQRST-complex.

**Comment by DominiqueMakowski** [comment]
Sorry, I'm not sure I understand what you mean by determining the PQRST complex from within the simulator, could you elaborate a bit more?

**Reply by nachoXXL** [reply]
The PQRST complex as shown in this image [https://www.researchgate.net/publication/328097407/figure/fig1/AS:678365954600962@1538746455334/PQRST-complex-of-a-normal-lead-II-ECG-recording.ppm](PQRST-complex), thus P-onset, P-peak, P-offset etc. These values can be determined with neurokit's ecg_delineate. But as said this fails in some perspective.
I can imagine that these values also can be determined from the simulator directly. I'm not familiar with wavelets but, maybe it is possible the extract/calculate these values eg from the wavelets directly. This might be possible eg if there is a wavelet for each part of said complex.
